plugin/evm/validators_cache_test.go

```diff
@@ -0,0 +1,84 @@
+package evm
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/flare/ids"
+)
+
+type testData map[uint64]map[ids.ShortID]uint64
+
+func (testData) ByEpoch(uint64) (map[ids.ShortID]uint64, error) {
+	return map[ids.ShortID]uint64{}, nil
+}
+
+func TestWithCacheSize(t *testing.T) {
+	wantCacheSize := uint(42)
+
+	cfg := CacheConfig{}
+	WithCacheSize(wantCacheSize)(&cfg)
+
+	assert.Equal(t, wantCacheSize, cfg.CacheSize)
+}
+
+func TestValidatorsCache_ByEpoch(t *testing.T) {
+	var (
+		idsList []ids.ShortID
+		weights []uint64
+
+		cacheTestData = testData{}
+	)
+
+	numValidators := 6
+	for i := 0; i < numValidators; i++ {
+		id := randID()
+		weight := rand.Uint64()
+
+		idsList = append(idsList, id)
+		weights = append(weights, weight)
+		epoch := uint64(0)
+		if i > numValidators / 2 {
+			epoch = uint64(1)
+		}
+		cacheTestData[epoch] = map[ids.ShortID]uint64{
+			id: weight,
+		}
+	}
+
+	validatorsCache := NewValidatorsCache(cacheTestData, WithCacheSize(32))
+	for k, v := range cacheTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+
+	t.Run("nominal case", func(t *testing.T) {
+		for i, epoch := range epochs {
+			result, err := validatorsCache.ByEpoch(epoch)
+			require.NoError(t, err)
+			pos := numValidators - 1
+			if i == 1 {
+				pos = (numValidators / 2) - 1
+			}
+			value, ok := result[idsList[pos]]
+			assert.True(t, ok)
+			want:= cacheTestData[epoch][idsList[pos]]
+			assert.Equal(t, want, value)
+		}
+	})
+
+	t.Run("handles missing key", func(t *testing.T) {
+		res, err := validatorsCache.ByEpoch(uint64(7))
+		require.NoError(t, err)
+		assert.Len(t, res, 0)
```

Comment:
This is not needed IMO. In Go when a function returns an error, its other return values should be assumed to be invalid data that could lead to undefined behavior (similar to how in C++ you're not supposed to be dereferencing null or dangling pointers).

In practice, in most cases, we just return a `nil` pointer or a zero value, but still because of that we don't need to check it in an error test IMO.